### PR TITLE
boards/nz32-sc151: add sx127x driver pin definition

### PR DESCRIPTION
**Pull request by fjmolinas**
Adds pin configuration to work with sx127x driver.

### CODE REVIEW

**Comment by fjmolinas** [comment]
@aabadie I have a doubt about how the sx127x_params file should work. I feel like it shouldn't use the SX127X_BOARD_PARAMS but redefine the SX127x_DEFAULT_PARAMS to leave the first one for specific user defining. And in that case, should this de defined threw auto-init or just included in the board.h.

**Comment by aabadie** [comment]
normally, auto_init is supposed to include `sx127x_params.h`. The `main.c` in `tests/drivers_sx127x` does the same.
If you copy the file `sx127x_params.h` from `drivers/sx127x/include` into `tests/drivers_sx127x`, the one in application directory will be included first and the drivers one will be ignored because of header guards. This way you can overwrite the default configuration of the driver from your application code.
The same strategy can be applied in boards include directory because it's also included before the drivers include.
What I would do with this PR is add a `sx127x_params.h` file in `boards/nz32-l151/include` containing:
```c
/* Doxygen stuff */

#ifndef SX127X_PARAMS_H
#define SX127X_PARAMS_H

#ifdef __cplusplus
extern "C" {
#endif

/**
 * @name    Set default configuration parameters for the SX127X driver
 * @{
 */
#ifndef SX127X_PARAM_SPI
#define SX127X_PARAM_SPI                    (SPI_DEV(0))
#endif

#ifndef SX127X_PARAM_SPI_SPEED
#define SX127X_PARAM_SPI_SPEED              (SPI_CLK_1MHZ)
#endif

#ifndef SX127X_PARAM_SPI_MODE
#define SX127X_PARAM_SPI_MODE               (SPI_MODE_0)
#endif

#ifndef SX127X_PARAM_SPI_NSS
#define SX127X_PARAM_SPI_NSS                GPIO_PIN(PORT_C, 8)
#endif

#ifndef SX127X_PARAM_RESET
#define SX127X_PARAM_RESET                  GPIO_PIN(PORT_A, 9)
#endif

#ifndef SX127X_PARAM_DIO0
#define SX127X_PARAM_DIO0                   GPIO_PIN(PORT_B, 0)
#endif

#ifndef SX127X_PARAM_DIO1
#define SX127X_PARAM_DIO1                   GPIO_PIN(PORT_B, 1)
#endif

#ifndef SX127X_PARAM_DIO2
#define SX127X_PARAM_DIO2                   GPIO_PIN(PORT_C, 6)
#endif

#ifndef SX127X_PARAM_DIO3
#define SX127X_PARAM_DIO3                   GPIO_PIN(PORT_A, 10)
#endif
/**@}*/

/**
 * @brief   SX127X configuration
 */
static const sx127x_params_t sx127x_params[] =
{ .spi       = SX127X_PARAM_SPI,     \
  .nss_pin   = SX127X_PARAM_SPI_NSS, \
  .reset_pin = SX127X_PARAM_RESET,   \
  .dio0_pin  = SX127X_PARAM_DIO0,    \
  .dio1_pin  = SX127X_PARAM_DIO1,    \
  .dio2_pin  = SX127X_PARAM_DIO2,    \
  .dio3_pin  = SX127X_PARAM_DIO3 }
};

#ifdef __cplusplus
}
#endif

#endif /* SX127X_PARAMS_H */
```

**Comment by aabadie** [comment]
Only board specific hardware definitions should go in `board.h`: on-board led, on-board button, integrated peripherals (SPI SD card, SPI radio, etc). The nz32-l151 LoRa module is external to this board, its definitions need to be an extra file.

**Comment by fjmolinas** [comment]
Yes, I had gotten that part from the other PR and you are totally right. I had an error with the way I was comipiling and that's why I wrongly thought I might also need autonit. Anyway, I addressed your comments.

**Comment by aabadie** [comment]
Thanks, I'm fine with your changes now. Could you amend the commit message as well ?
something like `boards/nz2-sc151: add inAir9 LoRa module configuration`.
Maybe @kYc0o can try this PR ?

**Comment by fjmolinas** [comment]
@aabadie done

File: boards/nz32-sc151/include/sx127x_params.h [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2016 Fundacion Inria Chile
```
Comment (aabadie) [inline boards/nz32-sc151/include/sx127x_params.h]: 2017

File: boards/nz32-sc151/include/sx127x_params.h [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  */
+ static const sx127x_params_t sx127x_params[] =
+ {
+ #ifdef SX127X_PARAMS_BOARD
+     SX127X_PARAMS_BOARD,
+ #else
+     SX127X_PARAMS_DEFAULT,
+ #endif
+ };
+
```
Comment (aabadie) [inline boards/nz32-sc151/include/sx127x_params.h]: Missing C++ compatibility code here:
```c
#ifdef __cplusplus
}
#endif
```

File: boards/nz32-sc151/include/sx127x_params.h [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ #endif
```
Comment (aabadie) [inline boards/nz32-sc151/include/sx127x_params.h]: Missing header guard close comment: 
```
/* SX127X_PARAMS_H */
```

Plus add the doxygen global block close comment:
```c
/** @} */
```
at the very end of the file

File: boards/nz32-sc151/include/sx127x_params.h (hunk 1 above)
Comment (fjmolinas) [inline boards/nz32-sc151/include/sx127x_params.h]: copy pasta, will do

**Comment by fjmolinas** [comment]
@aabadie addresed comments, not sure what murdock is complaining about now

**Comment by aabadie** [comment]
> not sure what murdock is complaining about now

Indeed, I don't understand the issue with header guard, they are ok. Can you just squash ? At least, this will fix the other issue.

**Comment by aabadie** [comment]
@fjmolinas, I have it ! There's an empty space at the beginning of each line in the file !! That's why the header guard check script cannot find the `#ifndef`... and fails.

**Comment by fjmolinas** [comment]
@aabadie squashed, and removed starting blank space.

**Comment by fjmolinas** [comment]
@haukepetersen Actually the spi defines are used, but in sx127x_internal.c file when spi_acquire(..) is called.

`spi_acquire(dev->params.spi, SPI_CS_UNDEF, SX127X_PARAM_SPI_MODE, SX127X_PARAM_SPI_SPEED);`

Therefore those defines must be kept in both files. I can change it to `/**
 * @name    Set default configuration parameters for the SX127X driver
 *          Pins are adapted to nz32-sc151 using modtronix inAir9 Lora
 *          module.
 * @{
 */

#ifndef SX127X_PARAM_SPI_SPEED
#define SX127X_PARAM_SPI_SPEED              (SPI_CLK_1MHZ)
#endif

#ifndef SX127X_PARAM_SPI_MODE
#define SX127X_PARAM_SPI_MODE               (SPI_MODE_0)
#endif

#define SX127X_PARAMS_DEFAULT     { .spi       = SPI_MODE_0, \
                                    .nss_pin   = GPIO_PIN(PORT_C, 8), \
                                    .reset_pin = GPIO_PIN(PORT_A, 9), \
                                    .dio0_pin  = GPIO_PIN(PORT_B, 0), \
                                    .dio1_pin  = GPIO_PIN(PORT_B, 1), \
                                    .dio2_pin  = GPIO_PIN(PORT_C, 6), \
                                    .dio3_pin  = GPIO_PIN(PORT_A, 10)  }
/**@}*/
`

**Comment by haukepetersen** [comment]
No, make it more simple as I stated above.

About the `SPEED` and `MODE` defines: those should not be part of the params file in the first place if they are not configured on a per-device basis! As they are used, these paramters are singletons used system wide, so they should be treated in that way.

Now, the SPI mode does not need to be configurable at all, as this is fixed for this class of devices -> define the mode locally in the source file.

For the SPI speed we have two options: 
- (a) make the speed configurable on a per device basis (as done for most spi devices in RIOT so far) -> add an `spi_clk` or similar parameter to the device params struct and assign the parameter in the `xxx_params.h` file(s)
- (b) use a shared speed value for ALL connected `sx127x` devices -> define a default value in the drivers interface header file (`drivers/include/sx127x.h`):
```c
#ifndef SX127X_SPI_CLK
#define SX127X_SPI_CLK       (SPI_CLK_1MHZ)
#endif
```

**Comment by aabadie** [comment]
Opened #7371 to address the comment above @haukepetersen.
So in the end, only 
```c
/**
 * @brief   Default configuration parameters for the SX127X driver
 *          Pins are adapted to nz32-sc151 using modtronix inAir9 Lora
 *          module.
 */
static const sx127x_params_t sx127x_params[] =
{
    {
        .spi       = SPI_DEV(0),
        .nss_pin   = GPIO_PIN(PORT_C,  8),
        .reset_pin = GPIO_PIN(PORT_A,  9),
        .dio0_pin  = GPIO_PIN(PORT_B,  0),
        .dio1_pin  = GPIO_PIN(PORT_B,  1),
        .dio2_pin  = GPIO_PIN(PORT_C,  6),
        .dio3_pin  = GPIO_PIN(PORT_A, 10) 
    }

};
```

**Comment by haukepetersen** [comment]
Oui oui :-)

**Comment by fjmolinas** [comment]
@haukepetersen Addressed changes, will require #7371 being merged.

File: boards/nz32-sc151/include/sx127x_params.h [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * @file
+ * @brief       sx127x configuration for nz32-sc15 using modtronix inAir9
+ *              LoRa module
+ * @author      Francisco Molina <francisco.molina@inria.cl>
+*/
+
+#ifndef SX127X_PARAMS_H
+#define SX127X_PARAMS_H
+
+#include "board.h"
```
Comment (haukepetersen) [inline boards/nz32-sc151/include/sx127x_params.h]: no need for including the board.h here anymore.

Comment (fjmolinas) [inline boards/nz32-sc151/include/sx127x_params.h]: yep, you are right, addressed.

**Comment by aabadie** [comment]
@haukepetersen, this one is ready, just missing your ACK

**Comment by haukepetersen** [comment]
all green -> lets go.

**Pull request merged**